File: tests/dynamic_fixtures/rust/xss_negative.rs
/// XSS — negative fixture.
///
/// Safe function: HTML-escapes user input before embedding in output.
/// Expected verdict: NotConfirmed (XSS payload is escaped; no raw script tag in output).
/// Cap: HTML_ESCAPE  Entry: `run(payload: &str)`
pub fn run(payload: &str) {
    // Safe: escape all HTML special characters before rendering.
    let escaped = payload
        .replace('&', "&amp;")
        .replace('<', "&lt;")
        .replace('>', "&gt;")
        .replace('"', "&quot;")
        .replace('\'', "&#x27;");
    let html = format!("<div class='comment'>{}</div>", escaped);
    println!("{}", html);
}

File: tests/dynamic_fixtures/python/xss_negative.py
"""XSS — negative fixture.

Safe function: uses html.escape() before rendering.
Expected verdict: NotConfirmed (script tag escaped to &lt;script&gt;).
"""
import html


def render_comment(user_input):
    """Safe: HTML-escapes user input before rendering."""
    safe = html.escape(user_input)
    print(f"<div class='comment'>{safe}</div>")
